### PR TITLE
Merge CLI into main

### PR DESCRIPTION
**Pull request by timvdlee**

### CODE REVIEW

**Comment by Timmerman73** [comment]
CLI is considered finished and ready for review.

File: FSHD_CLI/FSHD/FSHD.py [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+        print(f"No results found for this selection {' '.join(str(i) for i in args.selection)} for population {args.population}!\n"
+              f"For more information use the -h flag.")
+
+
+def main():
+    args, parser = parser_args()
+    if args.selection and len(args.selection) != 4:
+        parser.error(
+            f"You must have a input of 4 SSLP's. The length of your input is {len(args.selection)} SSLP's.")
+    if validate_combinations(args):
```
Comment (ifokkema) [inline FSHD_CLI/FSHD/FSHD.py:331]: The function lets me do:
`python3 FSHD.py -s 1 2 3 4 -p African -A haplotypes.json`
I was looking for the documented method of providing an alternative JSON file, but it was trying to add `haplotypes.json` to itself.
Valid combinations are a nice way to check, but you also need to exclude certain combinations.

File: FSHD_CLI/README.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
+
+
+
+# Haplotype Predictor
+
+This Python script allows you to predict genotypes from Short Sequence Length Polymorphisms (SSLPs) values for Facioscapulohumeral muscular dystrophy (FSHD) analysis. It requires a selection of SSLPs and the region from where the sample is derived.
```
Comment (ifokkema) [inline FSHD_CLI/README.md:6]: region=population :wink:
I suggest: "(...) the population to which the individual belongs, from where the sample is derived."

File: FSHD_CLI/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+## Pip package
+This analayser can also be used as a standalone pip package. It can be installed by navigating to the FSHD_CLI directory and executing the following command in your enviroment of choice.
```
Comment (ifokkema) [inline FSHD_CLI/README.md:9]: You need a spell-checker :wink:

File: FSHD_CLI/README.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    - shutil
+    - argparse
+    - pkg_resources
+    - questionary
+    - pathlib
+    - itertools
+
+## Script Description
+
+The script uses pre-defined haplotypes for prediction. A haplotype is a group of genes within an organism that was inherited together from a single parent. This script uses haplotypes of certain populations (European, African, Asian) to predict the genotypes from the provided SSLP values.
```
Comment (ifokkema) [inline FSHD_CLI/README.md:35]: This definition of a haplotype isn't fully correct - please have a look.
Also, the script is actually trying to predict the exact haplotypes present in the patient. Genotypes isn't wrong, but I don't feel the sentence fully explains the script's function.

File: FSHD_CLI/README.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+You can also provide an optional input JSON file that contains the haplotypes. If no file is provided, the script uses default haplotypes.
```
Comment (ifokkema) [inline FSHD_CLI/README.md:37]: I didn't find this feature; are you sure it's present?

File: FSHD_CLI/README.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+2. **-s, --selection and -p, --population**: Use this to run the genotype predictor on the given SSLP values for a specific population. Example usage: -s 159 161 163 166 -p Asian.
+3. **-H, --haplotypes and -p, --population**: This combination displays all possible haplotypes for a given population. Example usage: -H -p Asian.
+4. **-H, --haplotypes and -l, --list_datasets**: Use this combination to display all possible haplotypes and the list of all populations that have a haplotype dataset. Example usage: -H -l.
+5. **-A, --add**: Use this argument independently to add new haplotype files to the database. Example usage: -A ./new_haplotypes.json.
+## Example
+
+### shell
+
+```bash
+Python FSHD.py -s 159 161 163 166 -p European -n 10
```
Comment (ifokkema) [inline FSHD_CLI/README.md:67]: This wouldn't run because of the capital P in `Python`.

File: FSHD_CLI/README.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    - itertools
+
+## Script Description
+
+The script uses pre-defined haplotypes for prediction. A haplotype is a group of genes within an organism that was inherited together from a single parent. This script uses haplotypes of certain populations (European, African, Asian) to predict the genotypes from the provided SSLP values.
+
+You can also provide an optional input JSON file that contains the haplotypes. If no file is provided, the script uses default haplotypes.
+Usage
+
+shell
```
Comment (ifokkema) [inline FSHD_CLI/README.md:40]: Markup is missing here - this is supposed to be a header, right?

File: FSHD_CLI/README.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+## Example
+
+### shell
+
+```bash
+Python FSHD.py -s 159 161 163 166 -p European -n 10
+```
+
+This command will predict the genotypes for SSLP values 159 161 163 166 from the European region and display the top 10 results.
+Output
```
Comment (ifokkema) [inline FSHD_CLI/README.md:71]: Markup is missing here - this is supposed to be a header, right?

File: FSHD_CLI/README.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+ - chr10_1
+ - chr10_2
+ - Probability(%)
+ - Permissive
+ - Incidence(%)
+
+It also calculates and displays the Total Likelihood value at the end. If the -o/--output argument is used, the results will be written to the specified file instead of printing on the console.
+
+If no results are found for the given selection and region, a relevant message is displayed.
+Note
```
Comment (ifokkema) [inline FSHD_CLI/README.md:85]: Markup is missing here - this is supposed to be a header, right?

**Review by ifokkema — CHANGES_REQUESTED** [review]
Some changes are required - please see my comments in the code.
Also, a general comment; the script should always output at least one empty line before it exits; otherwise, the terminal will do something weird.

File: FSHD_CLI/README.md (hunk 2 above)
Comment (ifokkema) [reply FSHD_CLI/README.md]: Fixed in aaf66dc.

File: FSHD_CLI/README.md (hunk 3 above)
Comment (ifokkema) [reply FSHD_CLI/README.md]: Fixed in aaf66dc.

File: FSHD_CLI/README.md (hunk 5 above)
Comment (ifokkema) [reply FSHD_CLI/README.md]: Fixed in aaf66dc.

File: FSHD_CLI/README.md (hunk 6 above)
Comment (ifokkema) [reply FSHD_CLI/README.md]: Fixed in aaf66dc.

File: FSHD_CLI/README.md (hunk 8 above)
Comment (ifokkema) [reply FSHD_CLI/README.md]: Fixed in aaf66dc.

File: FSHD_CLI/README.md (hunk 9 above)
Comment (ifokkema) [reply FSHD_CLI/README.md]: Fixed in aaf66dc.

**Pull request merged**